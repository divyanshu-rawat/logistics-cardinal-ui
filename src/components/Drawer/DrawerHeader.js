import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

import DrawerHeaderStyled from './DrawerHeaderStyled';
import DrawerCloseButton from './DrawerCloseButton';

export const DrawerHeaderElement = styled('div')`
  flex: 1;
`;

export const DrawerHeaderButtonStyled = styled('div')`
  width: 50px;
  text-align: center;
`;

const DrawerHeader = ({ header, onClose }) => (
  <DrawerHeaderStyled>
    <DrawerHeaderElement>{header}</DrawerHeaderElement>
    <DrawerHeaderButtonStyled>
      <DrawerCloseButton onClose={onClose} />
    </DrawerHeaderButtonStyled>
  </DrawerHeaderStyled>
);

DrawerHeader.propTypes = {
  onClose: PropTypes.func.isRequired,
  header: PropTypes.node.isRequired,
};

export default DrawerHeader;
