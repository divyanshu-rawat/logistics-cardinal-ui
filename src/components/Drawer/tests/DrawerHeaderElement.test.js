import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import { DrawerHeaderElement } from '../DrawerHeader';

const renderedComponent = () => shallow(<DrawerHeaderElement />);

describe('<DrawerHeaderElement />', () => {
  it('should render the component', () => {
    const tree = toJson(renderedComponent());
    expect(tree).toMatchSnapshot();
  });
});
