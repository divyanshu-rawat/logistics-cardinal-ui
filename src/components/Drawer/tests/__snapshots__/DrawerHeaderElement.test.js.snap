// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DrawerHeaderElement /> should render the component 1`] = `
<StyledComponent
  forwardedClass={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "DrawerHeader__DrawerHeaderElement-brJxYn",
        "isStatic": true,
        "rules": Array [
          "flex:1;",
        ],
      },
      "displayName": "DrawerHeader__DrawerHeaderElement",
      "render": [Function],
      "styledComponentId": "DrawerHeader__DrawerHeaderElement-brJxYn",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
/>
`;
